core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/TransactionRuntimeExecutionExceptionMapper.java

```diff
@@ -32,19 +32,28 @@ public Response toResponse(TransactionRuntimeExecutionException exception) {
 
     @Override
     Status getResponseStatus() {
+        TransactionRuntimeExecutionException deepest = getDeepestTransactionException(CURRENT_EXCEPTION.get());
         return exceptionMappers.stream()
-            .filter(m -> m.getExceptionType().equals(CURRENT_EXCEPTION.get().getCause().getCause().getClass()))
+            .filter(m -> m.getExceptionType().equals(deepest.getCause().getCause().getClass()))
             .map(BaseExceptionMapper::getResponseStatus)
             .findFirst().orElse(Status.INTERNAL_SERVER_ERROR);
     }
 
+    private TransactionRuntimeExecutionException getDeepestTransactionException(TransactionRuntimeExecutionException e) {
+        if (e.getCause().getCause() instanceof TransactionRuntimeExecutionException) {
```

Comment:
Because the method is tail recursive would be better to swap statement and call getDeepestTransactionException() at the tail of the method.

Reply:
ok